workflow/controller/operator.go

```diff
@@ -1022,7 +1031,7 @@ func (woc *wfOperationCtx) processNodeRetries(node *wfv1.NodeStatus, retryStrate
 			retryOnFailed = true
 			retryOnError = true
 		}
-	case wfv1.RetryPolicyOnFailure, "":
+	case wfv1.RetryPolicyOnFailure:
```

Comment:
I noticed some conditional logic on line 2018 that checks the value of `retryStrategy.RetryPolicy`. Does that need to be updated?

Reply:
Good catch, I've done that section too.